### PR TITLE
Add EKS cluster create/teardown tasks

### PR DESCRIPTION
**Pull request by theofpa**
# Changes

Adds `eks-cluster-create` and `eks-cluster-teardown` tasks with the `eksctl` tool using the official image of weaveworks.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by vinamra28** [comment]
hi @theofpa. in order to add yourself as the owner of the task you need to be a member of the org. Probably you can follow the process just like https://github.com/tektoncd/community/pull/242

/ok-to-test

File: task/eks-cluster-create/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# EKS Cluster Create
+
+The Task `eks-cluster-create` can be used to create an EKS cluster in an AWS account, using the `eksctl` command,
+and fetch a kubeconfig that can be used (in a context with both kubectl and aws credentials available) to make 
+requests to the cluster.
+
+## Parameters
+
+* **cluster-name**: The name of the EKS cluster which will be teared down. (_required_)
```
Comment (vinamra28) [inline task/eks-cluster-create/0.1/README.md:9]: I guess this is what you meant...right?
```suggestion
* **cluster-name**: The name of the EKS cluster which you want to spin. (_required_)
```

File: task/eks-cluster-create/0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+* **cluster-version**: The EKS version to install. (_default_:1.17
```
Comment (vinamra28) [inline task/eks-cluster-create/0.1/README.md:10]: nit:
```suggestion
* **cluster-version**: The EKS version to install. (_default_:1.17)
```

File: task/eks-cluster-create/0.1/eks-cluster-create.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  - name: nodegroup-name
+    default: linux-nodes
+  - name: node-type
+    default: m5.xlarge
+  - name: desired-nodes
+    default: "4"
+  - name: min-nodes
+    default: "1"
+  - name: max-nodes
+    default: "4"
```
Comment (vinamra28) [inline task/eks-cluster-create/0.1/eks-cluster-create.yaml:34]: can we add a short description for the params as well over here? Probably you can take reference from [here](https://github.com/tektoncd/catalog/blob/6e8b8197343d78c397d997d26aae44b19c1ea266/task/aws-cli/0.1/aws-cli.yaml#L21)

File: task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    tekton.dev/displayName: "EKS Cluster Teardown"
+spec:
+  description: |
+    Teardown an EKS cluster.
+
+    This Task can be used to teardown an EKS cluster in an AWS account.
+  params:
+  - name: cluster-name
+  - name: region
+    default: us-west-2
```
Comment (vinamra28) [inline task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml:19]: same as above

**Review by vinamra28 — CHANGES_REQUESTED** [review]
can you please squash the commits as well :)

File: task/eks-cluster-create/0.1/eks-cluster-create.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  workspaces:
+  - name: secrets
+    mountPath: /tekton/home/.aws
+    description: The secret with the AWS keys
+  - name: kubeconfig
+    description: |
+      A workspace into which a kubeconfig file called `kubeconfig` will be written that will contain the information required to access the cluster. The `kubeconfig` will expect to use [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator/) to authenticate, so in order for it to be used it must be run in a container which contains both `kubectl` and `aws-iam-authenticator`.
+  steps:
+  - name: write-kubeconfig
+    image: weaveworks/eksctl:0.35.0
```
[the remark this answers is not in the document]
Comment (theofpa) [reply task/eks-cluster-create/0.1/eks-cluster-create.yaml]: fixed in cc49d35

File: task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  workspaces:
+  - name: secrets
+    mountPath: /tekton/home/.aws
+    description: The service account with the AWS keys
+  steps:
+  - name: delete-cluster
+    image: weaveworks/eksctl:0.35.0
```
[the remark this answers is not in the document]
Comment (theofpa) [reply task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml]: fixed in cc49d35

File: task/eks-cluster-create/0.1/README.md (hunk 1 above)
Comment (theofpa) [reply task/eks-cluster-create/0.1/README.md]: fixed in cc49d35

File: task/eks-cluster-create/0.1/README.md (hunk 2 above)
Comment (theofpa) [reply task/eks-cluster-create/0.1/README.md]: fixed in cc49d35

File: task/eks-cluster-create/0.1/eks-cluster-create.yaml (hunk 3 above)
Comment (theofpa) [reply task/eks-cluster-create/0.1/eks-cluster-create.yaml]: fixed in cc49d35

File: task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml (hunk 4 above)
Comment (theofpa) [reply task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml]: fixed in cc49d35

**Comment by theofpa** [comment]
> can you please squash the commits as well :)

Thanks for the review @vinamra28! I've addressed your comments in cc49d35.

File: task/eks-cluster-create/0.1/eks-cluster-create.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  workspaces:
+  - name: secrets
+    mountPath: /tekton/home/.aws
+    description: The secret with the AWS keys
+  - name: kubeconfig
+    description: |
+      A workspace into which a kubeconfig file called `kubeconfig` will be written that will contain the information required to access the cluster. The `kubeconfig` will expect to use [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator/) to authenticate, so in order for it to be used it must be run in a container which contains both `kubectl` and `aws-iam-authenticator`.
+  steps:
+  - name: write-kubeconfig
+    image: weaveworks/eksctl:0.35.0@sha256:48c1fa508970a01fd87a73ac7932a7160479d678cd019a3c84533d911fc54327
```
Comment (vinamra28) [inline task/eks-cluster-create/0.1/eks-cluster-create.yaml:53]:
```suggestion
    image: docker.io/weaveworks/eksctl:0.35.0@sha256:48c1fa508970a01fd87a73ac7932a7160479d678cd019a3c84533d911fc54327
```

File: task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  - name: region
+    default: us-west-2
+    description: The region where the cluster is in.
+  workspaces:
+  - name: secrets
+    mountPath: /tekton/home/.aws
+    description: The service account with the AWS keys
+  steps:
+  - name: delete-cluster
+    image: weaveworks/eksctl:0.35.0@sha256:48c1fa508970a01fd87a73ac7932a7160479d678cd019a3c84533d911fc54327
```
Comment (vinamra28) [inline task/eks-cluster-teardown/0.1/eks-cluster-teardown.yaml:28]:
```suggestion
    image: docker.io/weaveworks/eksctl:0.35.0@sha256:48c1fa508970a01fd87a73ac7932a7160479d678cd019a3c84533d911fc54327
```

**Review by vinamra28 — CHANGES_REQUESTED** [review]

**Comment by vinamra28** [comment]
@theofpa have you joined the org by accepting the invite you might have received via email?

**Comment by theofpa** [comment]
> @theofpa have you joined the org by accepting the invite you might have received via email?

oops, I've just done it, thanks!

**Comment by chmouel** [comment]
/lgtm
/approve

Thanks 👍🏼🔥

**Pull request merged**